test/DynamoCoreTests/DSCoreDataTests.cs

```diff
@@ -556,22 +556,345 @@ public void RememberWillUpdateCacheWhenPassedSupportedInputAndEmptyCacheJson()
+            var vInt = 5;
+            var vDouble = 3.14;
+            var vDateTime = DSCore.DateTime.ByDate(2001, 1, 1);
+            var vLocation = Location.ByLatitudeAndLongitude(50, 50);
+            var vTimeSpan = DSCore.TimeSpan.ByDateDifference(vDateTime, DSCore.DateTime.ByDate(2000, 1, 1));
+
+            var invStringList = new ArrayList() { vString, vInt, vDouble };
+            var vStringList = new ArrayList() { vString };
+            var vIntList = new ArrayList() { vInt };
+            var vDoubleList = new ArrayList() { vDouble };
+            var vDateTimeList = new ArrayList() { vDateTime };
+            var vLocationList = new ArrayList() { vLocation };
+            var vTimeSpanList = new ArrayList() { vTimeSpan };
+
+            // Assert - check list - succeed
+            var singleChecks = new Dictionary<Type, object>();
+            singleChecks[typeof(string)] = vString;
+            singleChecks[typeof(int)] = vInt;
+            singleChecks[typeof(double)] = vDouble;
+            singleChecks[typeof(DateTime)] = vDateTime;
+            singleChecks[typeof(Location)] = vLocation;
+            singleChecks[typeof(TimeSpan)] = vTimeSpan;
+
+            foreach (var kv in singleChecks)
+            {
+                Assert.AreEqual(
+                    true,
+                    DSCore.Data.IsSupportedDataType(kv.Value, kv.Key, false),
+                    String.Format($"Couldn't validate {kv.Key} input."));
+            }
+
+            // Assert - check list - fail
+            var validate = DSCore.Data.IsSupportedDataType(vStringList, typeof(string), false);
+            Assert.AreEqual(false, validate, "Shouldn't validate list values with list flag off.");
+
+            validate = DSCore.Data.IsSupportedDataType(vString, typeof(string), true);
+            Assert.AreEqual(false, validate, "Shouldn't validate single values with list flag on.");
+
+            validate = DSCore.Data.IsSupportedDataType(invStringList, typeof(string), true);
+            Assert.AreEqual(false, validate, "Shouldn't validate heterogenous list input.");
+
+            // Assert - check homogenous list values - succeed
+            var listChecks = new Dictionary<Type, object>();
+            listChecks[typeof(string)] = vStringList;
+            listChecks[typeof(int)] = vIntList;
+            listChecks[typeof(double)] = vDoubleList;
+            listChecks[typeof(DateTime)] = vDateTimeList;
+            listChecks[typeof(Location)] = vLocationList;
+            listChecks[typeof(TimeSpan)] = vTimeSpanList;
+
+            foreach (var kv in listChecks)
+            {
+                Assert.AreEqual(
+                    true,
+                    DSCore.Data.IsSupportedDataType(kv.Value, kv.Key, true),
+                    String.Format($"Couldn't validate {kv.Key} list input."));
+            }
+        }
+
+
+        [Test]
+        [Category("UnitTests")]
+        public void IsSupportedGeometryDataType()
```

Comment:
I wonder at the exhaustiveness of these tests. Do we really need to check that every type in the dropdown validates? Is it useful to keep the full list of supported types here _and_ in the dropdown items definition? Why isn't it enough to just check that the characteristics of the system work - test inheritance, is list/is not list, a type we know is not in the dropdown list, heterogenous list values?
Caveat: I don't have a lot of experience with TDD. Just seems like a lot of effort here for not a lot of validation value. @mjkkirschner?